### PR TITLE
Validate jax version for the both jit interface

### PR DESCRIPTION
**Pull request by rmoyard**
**Context:**
Jax update `0.4.4` brought on the staged processing for jitting (initial style), more details here https://jax.readthedocs.io/en/latest/autodidax.html#on-the-fly-final-style-and-staged-initial-style-processing

**Description of the Change:**

It creates bugs in Pennylane when we jit an inner function, because our definition of trainable parameters is JVP tracers, which not the case anymore with staged processing. It still works when jit is applied on the outer most function. See discussion https://github.com/google/jax/discussions/14805

The forward pass of both jit interface also needs some update and therefore both jit interfaces are disabled for Jax > 0.4.3.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

A potential solution for us is to introduce symbolic zeros in our custom_jvp function, but it is also not working for staged processing. Issue is opened there https://github.com/google/jax/issues/14833 and hopefully fixed soon.

See this PR where I tried to fix test for 0.4.5 https://github.com/PennyLaneAI/pennylane/pull/3868

### CODE REVIEW

File: pennylane/interfaces/jax_jit.py [hunk 1]
```diff
@@ -31,12 +31,10 @@
...
 def _validate_jax_version():
-    if semantic_version.match("<0.3.17", jax.__version__) or semantic_version.match(
-        "<0.3.15", jax.lib.__version__
-    ):
+    if semantic_version.match(">0.4.3", jax.__version__):
         msg = (
-            "The JAX JIT interface of PennyLane requires version 0.3.17 or higher for JAX "
-            "and 0.3.15 or higher JAX lib. Please upgrade these packages."
+            "The JAX JIT interface of PennyLane requires version between 0.4.1 and 0.4.3 for JAX "
+            "and JAX lib. Please upgrade downgrade these packages."
```
Comment (rmoyard) [inline pennylane/interfaces/jax_jit.py:37]:
```suggestion
            "and JAX lib. Please downgrade these packages."

```

File: pennylane/interfaces/jax_jit.py [hunk 2]
```diff
@@ -31,12 +31,10 @@
 
 
 def _validate_jax_version():
-    if semantic_version.match("<0.3.17", jax.__version__) or semantic_version.match(
```
Comment (timmysilv) [inline pennylane/interfaces/jax_jit.py:34]: why don't you want to check minimum version anymore?

Comment (rmoyard) [reply pennylane/interfaces/jax_jit.py:34]: I think it not valid anymore, my guess is that we support 0.4.1/0.4.2 and 0.4.3 but let me double check the minimal version.

Comment (mudit2812) [reply pennylane/interfaces/jax_jit.py:34]: I agree with Matt. If the minimum version we support is 0.4.1 then we should also check `semantic_version.match("<0.4.1", jax.__version__)`

File: pennylane/interfaces/jax_jit_tuple.py [hunk 3]
```diff
@@ -17,16 +17,27 @@
...
 import pennylane as qml
 from pennylane.interfaces.jax import _compute_jvps
-from pennylane.interfaces.jax_jit import _numeric_type_to_dtype, _validate_jax_version
+from pennylane.interfaces.jax_jit import _numeric_type_to_dtype
+from pennylane.interfaces import InterfaceUnsupportedError
 
 dtype = jnp.float64
 
 
+def _validate_jax_version():
```
Comment (mudit2812) [inline pennylane/interfaces/jax_jit_tuple.py:32]: Why is there a copy of this function in this file? The only difference I notice is the word "new" in the message.

**Review by mudit2812 — COMMENTED** [review]
Thanks for the changes Romain! I'm wondering if we should also check if the correct version of jaxlib is installed. There have been times when I get errors while running tests because of outdated versions of jax and jaxlib, but updating jax doesn't automatically update jaxlib as well.

File: pennylane/interfaces/jax_jit.py (hunk 2 above)
Comment (trbromley) [reply pennylane/interfaces/jax_jit.py:34]: So do we not support `>=0.3.17`?

**Comment by PhilipVinc** [comment]
Can I suggest using an actionable error like this one https://github.com/PennyLaneAI/pennylane/blob/c6c4e40e08cbfcf283c5e1233d39813cf4f2254a/pennylane/devices/default_qubit_jax.py#L44 ?
This tells users how to force-downgrade both jax and jaxlib (as many people don't know how to do that)

File: pennylane/interfaces/jax_jit_tuple.py (hunk 3 above)
Comment (rmoyard) [reply pennylane/interfaces/jax_jit_tuple.py:32]: I run into circular import issue if I import from the file

**Review by mudit2812 — APPROVED** [review]
Thanks for the changes Romain! 🚀

**Review by timmysilv — APPROVED** [review]

**Pull request merged**